outbound/converter/interchange_translator.py

```diff
@@ -49,20 +48,22 @@ def __pre_validate_segments(self):
 
     async def __generate_identifiers(self, sender, recipient):
         interchange_id, message_id, transaction_id = await asyncio.gather(
-            self.interchange_id_generator.generate_interchange_id(sender, recipient),
-            self.message_id_generator.generate_message_id(sender, recipient),
-            self.transaction_id_generator.generate_transaction_id()
+            self.id_generator.generate_interchange_id(sender, recipient),
+            self.id_generator.generate_message_id(sender, recipient),
+            self.id_generator.generate_transaction_id()
         )
         for segment in self.segments:
             if isinstance(segment, (InterchangeHeader, InterchangeTrailer)):
                 segment.sequence_number = interchange_id
-            if isinstance(segment, (MessageHeader, MessageTrailer)):
+            elif isinstance(segment, (MessageHeader, MessageTrailer)):
                 segment.sequence_number = message_id
-            if isinstance(segment, ReferenceTransactionNumber):
+            elif isinstance(segment, ReferenceTransactionNumber):
                 segment.reference = transaction_id
 
     def __translate_edifact(self):
         return '\n'.join([segment.to_edifact() for segment in self.segments])
 
-    async def __record_outgoing_state(self):
+    async def record_outgoing_state(self):
```

Comment:
This method should remain private (prefixed by __). There is no need for it to be called externally or for subclasses to override it.

Reply:
done